### PR TITLE
feat: Context Menu

### PR DESCRIPTION
**Pull request by huntabyte**
This PR adds the `Context Menu` builder function, which has the same functionality as the `Dropdown Menu`, but displays the menu at the pointer position and is triggered by a right click or long-press in the trigger inside the trigger element. 

**TODO:**
- [x] Figure out how to best share code between the various menus

### CODE REVIEW

File: src/lib/builders/context-menu/index.ts [hunk 1]
```diff
@@ -0,0 +1,1346 @@
...
+
+	const checkboxItemDefaults = {
+		checked: writable(false),
+	};
+
+	const checkboxItem = hiddenAction({
+		role: 'menuitemcheckbox',
+		tabindex: -1,
+		'data-orientation': 'vertical',
+		'data-melt-part': 'item',
```
Comment (TGlide) [inline src/lib/builders/context-menu/index.ts:409]: Shouldn't it be checkbox?

**Review by TGlide — APPROVED** [review]
Awesome, you're on fire! I do wonder about two behaviours:
- Do we need an arrow for context menu? Maybe it's a nice to have, but for the preview I feel we don't need it.
- I feel the default positioning should be to the bottom right of wjere we click the mouse, instead of the center: Screenshot 2023-06-28 at 21 52 33 (imaigne that the cursor is at the top-left corner of this box)

File: src/lib/builders/context-menu/index.ts (hunk 1 above)
Comment (huntabyte) [reply src/lib/builders/context-menu/index.ts]: Working on fixing this :)

**Comment by huntabyte** [comment]
> Awesome, you're on fire! I do wonder about two behaviours:
> 
> * Do we need an arrow for context menu? Maybe it's a nice to have, but for the preview I feel we don't need it.
> * I feel the default positioning should be to the bottom right of wjere we click the mouse, instead of the center: Screenshot 2023-06-28 at 21 52 33 (imaigne that the cursor is at the top-left corner of this box)

You are right about both of those!

I also just consolidated a lot of the `menu` code :)

**Comment by huntabyte** [comment]
@TGlide if you wouldn't mind taking a look at how I'm exporting/handling the types. It feels messy/hectic but I'm unsure right now what the most elegant way to handle it is so any opinions are appreciated!

**Comment by TGlide** [comment]
> @TGlide if you wouldn't mind taking a look at how I'm exporting/handling the types. It feels messy/hectic but I'm unsure right now what the most elegant way to handle it is so any opinions are appreciated!

They look fine to me! We could separate them into a types.ts, since the file is so big, but no big deal if not.

**Comment by TGlide** [comment]
Also, @huntabyte some merge conflicts :(

**Pull request merged**